src/openlmis-permissions/permission.service.js

```diff
@@ -264,24 +265,36 @@
          *
          * @param  {string}  rightName          the name of the right
          * @param  {string}  programId          the id of the program
-         * @param  {string}  supervisoryNodeId  the id of the supervisory node   
+         * @param  {string}  facilityId         the id of the facility
          * @return {Promise}                    the promise resolving to a boolean, true if user has role with the given
-         *                                      right for the given program and supervisory node, false otherwise, the
-         *                                      promise is rejected if checking right fails
+         *                                      right for the given program and facility supervisory node, 
+         *                                      false otherwise, the promise is rejected if checking right fails
          */
-        function hasRoleWithRightForProgramAndSupervisoryNode(rightName, programId, supervisoryNodeId) {
+        function hasRoleWithRightForProgramAndSupervisoryNode(rightName, programId, facilityId) {
             return $q
                 .all([
+                    this.requisitionGroupService.getAll(),
```

Comment:
what this line does ? I'm a bit afraid that it is trying to retrieve all requistiongroup. It can be a huge performance problem

Reply:
It retrieves all requsition groups. I have checked on production database how many requisition groups are stored there:
Angola: 58
Malawi: 135
Selv: 342